otaclient/app/ota_client.py

```diff
@@ -589,115 +615,189 @@ def status(self) -> wrapper.StatusResponseEcuV2:
 
-    async def dispatch_rollback(self, _: wrapper.RollbackRequestEcu):
-        """Dispatch OTA rollback execution to background thread.
+        return wrapper.UpdateResponseEcu(
+            ecu_id=self.ecu_id, result=wrapper.FailureType.NO_FAILURE
+        )
+
+    async def dispatch_rollback(
+        self, request: wrapper.RollbackRequestEcu
+    ) -> wrapper.RollbackResponseEcu:
+        # prevent rollback operation if otaclient is not started
+        if self._otaclient_inst is None:
+            return wrapper.RollbackResponseEcu(
+                ecu_id=self.ecu_id, result=wrapper.FailureType.UNRECOVERABLE
+            )
+
+        # check and acquire lock
+        if self._update_rollback_lock.locked():
+            logger.warning(
+                f"ongoing operation: {self.last_operation=}, ignore incoming {request=}"
+            )
+            return wrapper.RollbackResponseEcu(
+                ecu_id=self.ecu_id, result=wrapper.FailureType.RECOVERABLE
+            )
+
+        # immediately take the lock if not locked
+        await self._update_rollback_lock.acquire()
+        self.last_operation = wrapper.StatusOta.ROLLBACKING
+
+        async def _rollback_task():
+            if self._otaclient_inst is None:
+                return
 
-        Raises:
-            OTAClientBusy if otaclient is already executing update/rollback.
-        """
-        if self.update_rollback_lock.locked():
-            raise OTAClientBusy(f"ongoing operation: {self.last_operation=}")
-        else:  # immediately take the lock if not locked
-            await self.update_rollback_lock.acquire()
-            self.last_operation = wrapper.StatusOta.ROLLBACKING
-
-        async def _rollback():
-            """Background waiting for rollback to be finished and
-            release update_rollback lock."""
             try:
-                await self._run_in_executor(self._otaclient.rollback)
+                await self._run_in_executor(self._otaclient_inst.rollback)
+            except Exception:
+                pass  # error should be collected by otaclient, not us
             finally:
                 self.last_operation = None
-                self.update_rollback_lock.release()
+                self._update_rollback_lock.release()
 
         # dispatch to background
-        asyncio.create_task(_rollback())
+        asyncio.create_task(_rollback_task())
+
+        return wrapper.RollbackResponseEcu(
+            ecu_id=self.ecu_id, result=wrapper.FailureType.NO_FAILURE
+        )
 
     async def get_status(self) -> wrapper.StatusResponseEcuV2:
-        return await self._run_in_executor(self._otaclient.status)
+        # otaclient is not started due to boot control startup failed
+        if self._otaclient_inst is None:
+            return self._otaclient_startup_failed_status
+
+        # otaclient core started, query status from it
+        return await self._run_in_executor(self._otaclient_inst.status)
```

Comment:
If otaclient fails to start, status API will return the pre-built status API response, which is generated based on parsing boot controller startup failure/otaclient core startup failure.

Reply:
If otaclient is started, then use the status report generated by running otaclient instance.